File: 1lab/py/a.py
from collections import deque
def counter(x):
    st = deque()
    st.append(x)
    j = 1
    while(len(st) != x):
        st.appendleft(x - j)
        i = st[0]
        while (i > 0): 
            temp = st[len(st)-1]
            st.pop()
            st.appendleft(temp)
            i -= 1
        j += 1
    return st
list = []
for _ in range(int(input())):
    x = int(input())
    list.append(counter(x))
for i in list:
    while(len(i) != 0):
        print(i.popleft(),end=' ')
    print()

File: 1lab/a.cpp
#include <bits/stdc++.h>
using namespace std;

deque<int> counter(int x){
    deque <int> deq;
    deq.push_back(x);
    int j = 1;
    while(deq.size() != x){
        deq.push_front(x - j);
        int i = deq.front();
        while( i > 0){
            int temp = deq.back();
            deq.pop_back();
            deq.push_front(temp);
            i--;
        }
        j++;
    }
    return deq;
}
int main(){
    int n; cin >> n;
    vector <deque<int>> v;
    for (int i = 0; i < n; i++){
        int x; cin >> x;
        v.push_back(counter(x));
    }
    for (int i = 0;i < v.size(); i++){
        while (!v[i].empty()){
            cout<<v[i].front()<<" ";
            v[i].pop_front();
        }
        cout<<endl;

    }
}